VultisigApp/VultisigApp/Services/CacheService.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  CacheService.swift
+//  VultisigApp
+//
+//  Created by Enrique Souza Soares on 12/07/24.
+//
+
+import Foundation
+
+class CacheService {
+    
+    static let shared = CacheService()
+    
+    private let cache: URLCache
+    private let cacheTimeout: TimeInterval
+    
+    private init(cache: URLCache = URLCache.shared, cacheTimeout: TimeInterval = 120) {
+        self.cache = cache
+        self.cacheTimeout = cacheTimeout
+    }
+    
+    func getCachedData<T: Codable>(for key: String, type: T.Type) -> T? {
```

Comment:
check .returnCacheDataElseLoad cache policy. 
No need to cache anything manually, just setup this policy for requests you need

https://developer.apple.com/documentation/foundation/nsurlrequest/cachepolicy

Reply:
I've changed to use the cache policy with some expiration time as suggested.